contracts/proxy-deployer/src/contract_interactions.rs

```diff
@@ -95,6 +107,66 @@ pub trait ContractInteractionsModule: config::ConfigModule + pause::PauseModule
             .execute_on_dest_context()
     }
 
+    /// Use this endpoint to transfer the ownership
+    /// This is needed to properly update the stored data
+    #[endpoint(changeOwnerAddress)]
+    fn change_owner(
+        &self,
+        contract_address: ManagedAddress,
+        new_owner: ManagedAddress,
+        opt_orig_owner: OptionalValue<ManagedAddress>,
+    ) {
+        self.can_call_endpoint(Some(contract_address.clone()));
+        require!(
+            self.blockchain().is_smart_contract(&contract_address),
+            "Contract address is not a SC"
+        );
+        require!(
+            !self.blacklisted_deployers_list().contains(&new_owner),
+            "New owner is blacklisted"
+        );
+        let mut caller = self.blockchain().get_caller();
+        if caller == self.blockchain().get_owner_address() {
+            require!(opt_orig_owner.is_some(), "Must provide original owner");
+            caller = opt_orig_owner.into_option().unwrap();
```

Comment:
You already checked for some.

```suggestion
            caller = unsafe { opt_orig_owner.into_option().unwrap_unchecked() };
```

Reply:
Updated.